src/main/java/InputText.java

```diff
@@ -0,0 +1,38 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class InputText {
+
+    private static final String DEFAULT_SEPARATORS = ",|:";
+    private static final String CUSTOM_SEPARATOR_FINDERS = "//(.)\n(.*)";
+
+    private final String text;
+    private final Pattern pattern;
+
+    public InputText(String text) {
+        this.text = text;
+        this.pattern = Pattern.compile(CUSTOM_SEPARATOR_FINDERS);
+    }
+
+    public static InputText from(String text){
+        return new InputText(text);
+    }
+
+    public boolean isBlank() {
+        return this.text == null || this.text.isEmpty();
+    }
+
+    private boolean hasCustomSeparator(){
+        return pattern.matcher(this.text).matches();
+    }
+
+    public String[] split(){
+        if(hasCustomSeparator()){
```

Comment:
기존에 구현하신 코드가 충분히 가독성이 괜찮았던 것 같은데, 지금 코드로 변경하신 이유가 있을까요?
```suggestion
Matcher m = pattern.matcher(this.text);
if (m.find()) {
    String customDelimiter = m.group(1);
    return m.group(2).split(customDelimiter);
}
return this.text.split(DEFAULT_SEPARATORS);
```

Reply:
추후 확장성을 고려해서 메소드화 해보았는데 굳이 그럴 필요는 없었던거 같네요!
저는 평소에 코드를 설계 할 때 확장성을 고려해서 설계를 할 경우 가끔 오버엔지니어링인가 라는 의문이 들곤 하는데요
혹시 홍석님은 확장성에 대해 얼마나 고민하시는지 궁금합니다!

Reply:
지운님께서 말씀하신 확장성과 오버 엔지니어링 사이의 고민은 저도 늘 고민하게 되고, 참 답을 찾기 어렵더라구요.

저는 아래와 같은 방법을 사용하는데, 지운님도 지운님만의 방법을 고민해보시면 좋을 것 같아요.
1. 처음부터 확장성을 고려하지는 않고 작성한다.
2. 기능이 변경, 확장될 필요가 있을 때 기존의 코드를 리팩토링한다.
3. 단, 처음부터 변경의 방향성을 알 수 있는 경우(e.g., 사업 부서에서 다음 스텝으로 하고 싶은 아이템이 있는 경우, 개발 조직에서 논의했을 때 우서순위에 따라 뒤로 미뤄놓은 기능이 있는 경우)에는 변경을 반영하기 쉬운 방향으로 설계한다.

언젠가 정답을 처음부터 알 수 있는 때가 오면 좋겠다는 생각이 드네요. 😀